x-pack/plugins/enterprise_search/public/applications/enterprise_search_overview/components/product_selector/ingestion_selector.tsx

```diff
@@ -75,28 +77,52 @@ export const IngestionSelector: React.FC = () => {
         </EuiFlexItem>
         {productFeatures.hasWebCrawler && (
           <EuiFlexItem>
-            <IngestionCard
-              buttonLabel={i18n.translate(
-                'xpack.enterpriseSearch.ingestSelector.method.crawlerButtonLabel',
-                {
-                  defaultMessage: 'Crawl URL',
-                }
-              )}
-              buttonIcon={CrawlerIcon}
-              description={i18n.translate(
-                'xpack.enterpriseSearch.ingestSelector.method.crawler.description',
-                {
-                  defaultMessage:
-                    'Discover, extract, and index searchable content from websites and knowledge bases.',
-                }
-              )}
-              href={generatePath(ENTERPRISE_SEARCH_CONTENT_PLUGIN.URL + NEW_CRAWLER_PATH)}
-              isDisabled={crawlerDisabled}
-              logo={crawlerLogo}
-              title={i18n.translate('xpack.enterpriseSearch.ingestSelector.method.crawler', {
-                defaultMessage: 'Web Crawler',
-              })}
-            />
+            {crawlerDisabled ? (
```

Comment:
It is a small change so it can stay like this. It is possible to do the differences on prop level like this

```
<IngestionCard 
    buttonLabel={!crawlerDisabled? 
i18n.translate('xpack....', { defaultMessage: 'Crawl URL' }) : 
i18n.translate('xpack....', { defaultMessage: 'Source code' })
}
```

Or for the ones that are repeating, extract them into a variable


```
const CRAWLER_LABEL  = i18n.translate('xpack.enterpriseSearch.ingestSelector.method.crawler', {
                  defaultMessage: 'Web Crawler',
                })

// and use it like 

<IngestionCard
    title={CRAWLER_LABEL} ...

```

Reply:
Good tip, it will look like this:

```
<IngestionCard
              buttonLabel={
                crawlerDisabled
                  ? i18n.translate(
                      'xpack.enterpriseSearch.ingestSelector.method.sourceCodeButtonLabel',
                      {
                        defaultMessage: 'Source code',
                      }
                    )
                  : i18n.translate(
                      'xpack.enterpriseSearch.ingestSelector.method.crawler.description',
                      {
                        defaultMessage:
                          'Discover, extract, and index searchable content from websites and knowledge bases.',
                      }
                    )
              }
              buttonIcon={crawlerDisabled ? GithubIcon : CrawlerIcon}
              description={i18n.translate(
                'xpack.enterpriseSearch.ingestSelector.method.crawler.description',
                {
                  defaultMessage:
                    'Discover, extract, and index searchable content from websites and knowledge bases.',
                }
              )}
              href={
                crawlerDisabled
                  ? CRAWLER.github_repo
                  : generatePath(ENTERPRISE_SEARCH_CONTENT_PLUGIN.URL + NEW_CRAWLER_PATH)
              }
              isBeta={crawlerDisabled}
              logo={crawlerLogo}
              title={i18n.translate('xpack.enterpriseSearch.ingestSelector.method.crawler', {
                defaultMessage: 'Web Crawler',
              })}
            />
```